doc/security.md

```diff
@@ -1,16 +1,30 @@
 # Security
 ## Introduction
-Local communications over the UNIX socket happen over a cleartext HTTP
-socket and access is restricted by socket ownership and mode.
+LXD is a daemon running as root.
 
+Access to that daemon is only possible over a local UNIX socket by default.
+Through configuration, it's then possible to expose the same API over
+the network on a TLS socket.
+
+**WARNING**: Local access to LXD through the UNIX socket always grants
+full access to LXD. This includes the ability to attach any filesystem
+paths or devices to any container as well as tweaking all security
+features on containers. You should only give such access to someone who
+you'd trust with root access to your system.
```

Comment:
@stgraber what is the default permissions on this unix socket, I've been able to access the `lxc` client as non-root user without doing any specific on ubuntu 18.04. My user is part of the `adm` group though (i.e it can sudo root).

Reply:
On Ubuntu, it is `root:lxd` 0660 for both the deb and snap.

In the deb case, existing members of the `sudo` group are added to the `lxd` group on installation, similarly to what's done for other packages like libvirt.

In the snap case, there is no such logic and it's up to the local administrator to add members to the group.

Reply:
@stgraber ah yes on self-build systems its set the socket to sudo group:

```
srw-rw---- 1 root sudo 0 May 17 08:54 /var/lib/lxd/unix.socket
```

Reply:
That actually depends on how you start it, I'd expect this when you pass `--group sudo`, otherwise it would be `root:root` by default.

Reply:
@stgraber ah yes, I setup that alias a while back when I first started following a readme, that makes sense. So its a secure by default to only root.